### PR TITLE
Support Optional isPresent call in assertThat

### PR DESCRIPTION
**Pull request by shas19**
1. Support for `assertThat(optionalFoo.isPresent()).isTure` call
2. separated the `Name list` to the new utility class. So it can be used by multiple handlers.   

Added unit test

### CODE REVIEW

File: nullaway/src/main/java/com/uber/nullaway/handlers/Util.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+import org.checkerframework.dataflow.cfg.node.MethodInvocationNode;
+import org.checkerframework.dataflow.cfg.node.Node;
+
+/**
+ * A utility class that holds the names from the Table. Currently, {@link
+ * com.uber.nullaway.handlers.AssertionHandler} requires it, while {@link
+ * com.uber.nullaway.handlers.OptionalEmptinessHandler} uses it only when AssertionHandler is
+ * enabled.
+ */
+class Util {
```
Comment (msridhar) [inline nullaway/src/main/java/com/uber/nullaway/handlers/Util.java]: This class needs a better name.  Maybe `NameUtil` or `MethodNameUtil`?

File: nullaway/src/main/java/com/uber/nullaway/handlers/OptionalEmptinessHandler.java [hunk 2]
```diff
@@ -139,6 +137,29 @@ public boolean includeApInfoInSavedContext(AccessPath accessPath, VisitorState s
...
   }
 
+  private void updateIfAssertIsPresentTrueOnOptional(
+      MethodInvocationNode node, Types types, AccessPathNullnessPropagation.Updates bothUpdates) {
+    Node receiver = node.getTarget().getReceiver();
+    if (receiver instanceof MethodInvocationNode) {
+      MethodInvocationNode receiverMethod = (MethodInvocationNode) receiver;
+      Symbol.MethodSymbol receiverSymbol = ASTHelpers.getSymbol(receiverMethod.getTree());
+      if (util.isMethodAssertThat(receiverSymbol)) {
+        Node arg = receiverMethod.getArgument(0);
```
[the remark this answers is not in the document]
Comment (shas19) [reply nullaway/src/main/java/com/uber/nullaway/handlers/OptionalEmptinessHandler.java]: It should throw an `exception`. There are two instances where I am doing this. In both the cases, it seems that the arguments cannot be empty

File: nullaway/src/main/java/com/uber/nullaway/handlers/Util.java (hunk 1 above)
Comment (shas19) [reply nullaway/src/main/java/com/uber/nullaway/handlers/Util.java]: Yeah..renamed to `MethodNameUtil`, which sounds better.

File: nullaway/src/main/java/com/uber/nullaway/handlers/OptionalEmptinessHandler.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (shas19) [reply nullaway/src/main/java/com/uber/nullaway/handlers/OptionalEmptinessHandler.java]: I'm assuming `assertThat` will always have some input in it. Is this correct?

[the remark this answers is not in the document]
Comment (shas19) [reply nullaway/src/main/java/com/uber/nullaway/handlers/OptionalEmptinessHandler.java]: Yes

**Pull request merged**